processing/src/main/java/org/apache/druid/segment/IndexMergerV9.java

```diff
@@ -332,6 +348,174 @@ private void makeMetadataBinary(
+                                         .map(AggregatorFactory::getName)
+                                         .collect(Collectors.toList());
+
+
+      final List<DimensionMergerV9> mergers = new ArrayList<>();
+      final Map<String, ColumnFormat> columnFormats = Maps.newLinkedHashMapWithExpectedSize(dimensions.size() + metrics.size());
+
+      for (String dimension : dimensions) {
+        final ColumnFormat dimensionFormat = projectionAdapters.get(0).getFormat(dimension);
+        columnFormats.put(dimension, dimensionFormat);
+        DimensionHandler handler = dimensionFormat.getColumnHandler(dimension);
+        DimensionMergerV9 merger = handler.makeProjectionMerger(
+            Projections.getProjectionSmooshV9FileName(spec, dimension),
+            indexSpec,
+            segmentWriteOutMedium,
+            dimensionFormat.toColumnCapabilities(),
+            progress,
+            closer
+        );
+        if (parentMergers.containsKey(dimension)) {
+          merger.attachParent(parentMergers.get(dimension), projectionAdapters);
+        } else {
+          merger.writeMergedValueDictionary(projectionAdapters);
+        }
+        mergers.add(merger);
+      }
+      for (String metric : metrics) {
+        columnFormats.put(metric, projectionAdapters.get(0).getFormat(metric));
+      }
+
+
+      final GenericColumnSerializer timeWriter;
+      if (spec.getTimeColumnName() != null) {
+        timeWriter = setupTimeWriter(segmentWriteOutMedium, indexSpec);
+      } else {
+        timeWriter = null;
+      }
+      final ArrayList<GenericColumnSerializer> metricWriters =
+          // todo (clint): call static method to build column name
+          setupMetricsWriters(segmentWriteOutMedium, metrics, columnFormats, indexSpec, spec.getName() + ".");
+
+      Function<List<TransformableRowIterator>, TimeAndDimsIterator> rowMergerFn =
+          rowIterators -> new RowCombiningTimeAndDimsIterator(rowIterators, spec.getAggregators(), metrics);
+
+      List<TransformableRowIterator> perIndexRowIterators = Lists.newArrayListWithCapacity(projectionAdapters.size());
+      for (int i = 0; i < projectionAdapters.size(); ++i) {
+        final IndexableAdapter adapter = projectionAdapters.get(i);
+        TransformableRowIterator target = adapter.getRows();
+        perIndexRowIterators.add(IndexMerger.toMergedIndexRowIterator(target, i, mergers));
+      }
+      final TimeAndDimsIterator timeAndDimsIterator = rowMergerFn.apply(perIndexRowIterators);
+      closer.register(timeAndDimsIterator);
+
+      int rowCount = 0;
+      List<IntBuffer> rowNumConversions = new ArrayList<>(projectionAdapters.size());
+      for (IndexableAdapter adapter : projectionAdapters) {
+        int[] arr = new int[adapter.getNumRows()];
+        Arrays.fill(arr, INVALID_ROW);
+        rowNumConversions.add(IntBuffer.wrap(arr));
+      }
+
+      final String section = "walk through and merge rows";
```

Comment:
```suggestion
      final String section = "walk through and merge rows for projections";
```

Reply:
yea, still need to adjust a lot of these things, its sort of adapted from the regular flow since its pretty similar in a lot of ways